autometa/config/environ.py

```diff
@@ -51,42 +52,23 @@
 ]
 
 
-def which(program):
-    """Finds the full path for an executable and checks read permissions exist.
-
-    See: https://stackoverflow.com/a/377028
-
+def which(program, mode=os.X_OK):
+    """
+    Finds the full path for an executable and checks read permissions exist
 
     Parameters
     ----------
     program : str
         the program to check
+    mode : path like object, optional
+        permission mask passed to os.access(), by default os.X_OK
 
     Returns
     -------
-    str
-        the path if it was valid or an empty string if not
-        eg. </path/to/executable/> or ''
-
-    Raises
-    -------
-    ExceptionName
-        Why the exception is raised.
-
+    str or None
+        the path if it was valid, eg. </path/to/executable/> or None if not
     """
-    def is_exe(fpath):
-        return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
-    # check to see if program path is given or just the name
-    fpath, __ = os.path.split(program)
-    if fpath:
-        if is_exe(program):
-            return program
-    else:
-        for path in os.environ["PATH"].split(os.pathsep):
-            exe_file = os.path.join(path, program)
-            if is_exe(exe_file):
-                return exe_file
-    return ''
+    return shutil.which(program)
```

Comment:
I am not sure, why this is not showing up for you. I can commit again if you like. 
@WiscEvan are you also not able to see the last commit?

Reply:
Perhaps I am not understanding what you are trying to do. I expected there to be no `which` function at all, just a call to `shutil.which(program)` wherever there would have been a call to `which` before. Is that not what you are trying to do?

Reply:
It seems like I had misunderstood what you and @WiscEvan were trying to convey. I have completely removed the `which` function now and have now called `shutil.which` wherever `which` function was being called.